src/NodaTime/Calendars/WondrousYearMonthDayCalculator.cs

```diff
@@ -0,0 +1,333 @@
+﻿// Copyright 2014 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NodaTime.Calendars.Wondrous;
+using NodaTime.Utility;
+using System;
+
+namespace NodaTime.Calendars
+{
+    /// <summary>
+    ///     See <see cref="CalendarSystem.Wondrous" /> for details.
+    /// </summary>
+    internal sealed class WondrousYearMonthDayCalculator : YearMonthDayCalculator
+    {
+        private const int WondrousMinYear = 1;
+        private const int WondrousMaxYear = 1000; // current lookup tables are pre-calculated for a thousand years
+
+        private const int AverageDaysPer10Years = 3652; // Ideally 365.2425 per year...
+
+        private const int DaysInMonth = 19;
+        private const int MonthsInYear = 19;
+
+        private const int Month18 = 18;
+        private const int Month19 = 19;
+
+        public const int AyyamiHaMonth0 = WondrousCalendarHelper.AyyamiHaMonth0;
+
+        private const int UnixEpochDayAtStartOfYear1 = -45941;
+
+        internal WondrousYearMonthDayCalculator()
+            : base(WondrousMinYear,
+                WondrousMaxYear - 1,
+                AverageDaysPer10Years,
+                UnixEpochDayAtStartOfYear1)
+        { }
+
+        protected override int GetDaysFromStartOfYearToStartOfMonth(int year, int month)
+        {
+            var daysFromStartOfYearToStartOfMonth = DaysInMonth * (month - 1);
+
+            if (month == Month19)
+            {
+                daysFromStartOfYearToStartOfMonth += WondrousCalendarHelper.GetYearInfo(year).DaysInAyyamiHa;
+            }
+
+            return daysFromStartOfYearToStartOfMonth;
+        }
+
+        /// <param name="year"></param>
```

Comment:
I wouldn't worry about XML docs for these overrides, unless there's anything specific to add. (I'll remove them.)